pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1643,7 +1658,8 @@ func sendHardwareHealthProtobufByURL(ctx *getconfigContext, harwdareHealthURL st
 
 	data, err := proto.Marshal(HardwareHealth)
 	if err != nil {
-		log.Fatal("sendHardwareHealthProtobufByURL proto marshaling error: ", err)
+		log.Errorf("sendHardwareHealthProtobufByURL proto marshaling error: %v", err)
```

Comment:
Have we seen the log.Fatal fire? In theory this should only happen when there is a memory allocation failure AFAIU. But there could be other cases (such as invalid UTF-8 which would indicate type confusion between byte strings and character strings), so it would make sense to check what failure you've seen here.

Reply:
I have not seen that fire. But we know that fatal triggers a reboot, so I wanted to avoid it. Do you think we should keep log.Fatal?

Reply:
For things which should never(tm) happen is is IMNSHO better to do a fatal then log an error and drive on as if nothing happened.